code/game/objects/items/weapons/twohanded.dm

```diff
@@ -624,4 +624,67 @@
 	sharpness = IS_SHARP
 
 /obj/item/weapon/twohanded/bonespear/update_icon()
-		icon_state = "bone_spear[wielded]"
+	icon_state = "bone_spear[wielded]"
+
+/*
+ * Sky Bulge (Gae Bolg, tradtional dragoon lance from many FF games.)
+ */
+/obj/item/weapon/twohanded/skybulge	//Copy+paste job from bonespear because no explosions.
+	icon_state = "sky_bulge0"
+	name = "Sky Bulge"
+	desc = "A legendary stick with a very pointy tip. Looks to be used for throwing. And jumping. Can be stubborn if you throw too much." //TODO: Be funnier.
+	force = 10 //This thing aint for robusting.
+	w_class = WEIGHT_CLASS_BULKY
+	slot_flags = SLOT_BACK
+	force_unwielded = 10
+	force_wielded = 18					//Same as regular spear. This is a utility weapon.
+	throwforce = 24						//And that utility is throwing. 24 so it takes 5 hits instead of 4.
+	throw_speed = 4
+	embedded_impact_pain_multiplier = 0	//If you somehow embed this, it's not going to hurt.
+	armour_penetration = 15				//Same as Bone Spear
+	embed_chance = 0					//Would ruin the whole theme of the thing.
+	hitsound = 'sound/weapons/bladeslice.ogg'
+	attack_verb = list("attacked", "poked", "jabbed", "torn", "gored", "lanced") //Added lanced for flavour.
+	sharpness = IS_SHARP
+	var/maxdist = 16
+	var/throw_cooldown = 5				//Should equate to half a second. Also should hardly come up because of the do_after.
+
+/obj/item/weapon/twohanded/skybulge/update_icon()
+	icon_state = "sky_bulge[wielded]"
+
+/obj/item/weapon/twohanded/skybulge/throw_at()  //Throw cooldown and offhand-proofing.
+	if(throw_cooldown > world.time)
+		var/mob/User = thrownby
+		User.put_in_hands(src)
+		User.visible_message("<span class='warning'>But it doesn't leave [User]'s hand!")	//Always let the user know input went through and all that.
+		return
+	unwield(src)
+	..()
+
+/obj/item/weapon/twohanded/skybulge/throw_impact(atom/target) //Praise be the ratvar spear for showing me how to use this proc.
+	var/turf/Target = get_turf(target)
+	var/mob/User = thrownby
+	var/turf/Source = get_turf(thrownby)
+
+	if(Source.z == ZLEVEL_STATION && Source.z == Target.z)	//On station + same z check
```

Comment:
if(source.z != target.z)
       return
    if(source.z == ZLEVEL_STATION && get_dist(Target, Source) > maxdist)
       return
    if(do_after(User, 5, target = src, progress = 0))
       stuff

this way you don't duplicate the code